app/models/framework_response.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class FrameworkResponse < VersionedModel
+  validates :type, presence: true
+
+  belongs_to :framework_question
+  belongs_to :person_escort_record
+  has_many :dependents, class_name: 'FrameworkResponse',
+                        foreign_key: 'parent_id'
+
+  belongs_to :parent, class_name: 'FrameworkResponse', optional: true
+  validates_each :value, on: :update do |record, _attr, _value|
+    if record.parent&.option_selected?(record.framework_question.dependent_value) && record.framework_question.required
```

Comment:
Is it possible for the framework_question to be nil?

Reply:
We can never have a response without a question or a person escort record associated